### ISSUE TITLE
Test grid search capabilities over training hyperparameters

### ISSUE THREAD

**Issue by bethanyconnolly** [OpenBioML/chemnlp#252]
We want to check that our grid search functionality works and that we can train in parallel a range of models with different hyper-parameters and datasets. 
- [x] Identify model and data hyper-parameters to grid search over
- [ ] Create mixed datasets
- [x] Train models in parallel
- [x] Evaluate models and benchmark chemistry/nlp performance

**Comment by bethanyconnolly** [comment]
Tested this over [1, 3, 7]B models for 300M tokens of datasets [wikipedia, pubmed abstracts and elsevier] but not mixed together.
Grid search training and batched evaluation worked correctly.
Initial evaluation results [here](https://www.notion.so/stable-society/Grid-Search-_-300M-pure-data-sets-72c1b5b7f7eb4250b14d59aad45ee1b3)
Results on Lambada task are strange and coupled with the loss curves from the training runs, this points to an issue/error in the model training.

**Comment by jackapbutler** [comment]
reopening a new issue for this task as this was completed
